fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -1006,6 +1008,22 @@ compactionClause
     : (BASE | CUMULATIVE)? COMPACT (identifier | identifierList)?
     ;
 
+subfieldName
+    : identifier | '[*]'
```

Comment:
replace `'[*]` with ARRAY_ELEMETN?

Reply:
> replace `'[*]` with ARRAY_ELEMETN?

Done, changed in https://github.com/StarRocks/starrocks/pull/46619/commits/0003760e4df6dbf046756b8683af4a9a48ff75b9